src/black/trans.py

```diff
@@ -854,6 +857,11 @@ def _validate_msg(line: Line, string_idx: int) -> TResult[None]:
 
             set_of_prefixes.add(prefix)
 
+            if "f" in prefix and leaf.value[-1] != QUOTE:
```

Comment:
I changed it to only reject if it would both change the quote type and there are nested quotes in the string. That makes only 3 tests regress, instead of the many more before. I still feel like the ability to do this at all is a workaround, but I can also see not wanting to regress too much functionality.